syft/format/github/encoder.go

```diff
@@ -0,0 +1,46 @@
+package github
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/anchore/syft/syft/format/github/internal/model"
+	"github.com/anchore/syft/syft/sbom"
+)
+
+const ID sbom.FormatID = "github-json"
+
+type encoder struct {
+}
+
+func newFormatEncoder() sbom.FormatEncoder {
+	return encoder{}
+}
+
+func DefaultFormatEncoder() sbom.FormatEncoder {
```

Comment:
I'll need to think on this and other similar comments below... I think there is a problem about how the total set of encoders are derived and how that drives what's available on the CLI before running. I'll re-review and get back to you, because ultimately it would be ideal to not have these `Default` objects like you said (except configs of course).